spread.yaml

```diff
@@ -0,0 +1,54 @@
+project: chisel
+
+path: /chisel
+
+backends:
+  # Cannot use LXD backend due to https://github.com/snapcore/spread/issues/154
+  # lxd:
+  #   systems:
+  #     - ubuntu-bionic
+  #     - ubuntu-focal
+  #     - ubuntu-jammy
+  # GitHub actions (runners) don't support nested virtualization (https://github.com/community/community/discussions/8305)
+  # qemu:
+  #   systems:
+  #     - ubuntu-22.04:
+  #         username: ubuntu
+  #         password: ubuntu
+  #     - ubuntu-22.10:
+  #         username: ubuntu
+  #         password: ubuntu
+  adhoc:
+    allocate: |
+      echo "Allocating $SPREAD_SYSTEM..."
+      image=$(echo $SPREAD_SYSTEM | tr '-' ':')
+      docker pull $image
+      docker run -e usr=$SPREAD_SYSTEM_USERNAME -e pass=$SPREAD_SYSTEM_PASSWORD --name $SPREAD_SYSTEM -d $image sh -c '
+        set -x
+        apt update
+        apt install -y openssh-server sudo
+        mkdir /run/sshd /chisel
+        useradd -rm -d /home/ubuntu -s /bin/bash -g root -G sudo -u 1000 ubuntu
+        echo "$usr:$pass" | chpasswd
+        echo "$usr ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+        /usr/sbin/sshd -D
+      '
+      ADDRESS `docker inspect $SPREAD_SYSTEM --format '{{.NetworkSettings.Networks.bridge.IPAddress}}'`
+    discard:
+      docker rm -f $SPREAD_SYSTEM
+    systems:
+      - ubuntu-22.04:
+          username: ubuntu
+          password: ubuntu
+      - ubuntu-22.10:
```

Comment:
What's the benefit of having different host systems here?  Is that about testing the native Go release of each?

Reply:
Yes. It's about making sure Chisel works properly for every chisel-release in https://github.com/canonical/chisel-releases/

Reply:
testing only on 23.04 for now. We can address the multi-release tests in the chisel-releases repo later on